x-pack/plugins/security_solution/docs/prebuilt_rules_customization_rfc.md

```diff
@@ -0,0 +1,2862 @@
+
+**If a matching `rule_id` is found, but the `version` is not found**, it means there are some versions of this prebuilt rule known to Kibana, which means we should identify the rule being imported as prebuilt. The prebuilt rules package has a limit on the number of historical rule versions, and we can't assume that for a given rule_id we will always have ALL historical versions available as security-rule assets.
+
+In this case, we will the rule's params to be:
+```
+{ 
+  rule_source: {
+    type: 'external',
+    is_customized: false
+  },
+  immutable: true
+}
+```
+
+
+**If a matching `rule_id` is NOT found**, that means that the rule is a custom rule. And `rule_source` will be simply:
+```
+{
+  type: 'internal'
+}
+```
+
+And we can finally create a rule or update an existing rule using the request payload and the calculated `rule_source` field.
+
+--- 
+Given the requirements described above, the following table shows the behaviour of our endpoint for a combination of inputs and Kibana states.
+
+(All situation assume that the Elastic prebuilt rules package will be installed, since we have set this as a precondition of the endpoint)
+
+<table>
+  <thead>
+    <tr>
+      <th>Current state of rule<br>(match by rule_id)</th>
+      <th>Matching rule_id found as security-rule asset?</th>
+      <th>Matching version found for security-rule asset?</th>
+      <th>Example field in import payload</th>
+      <th>Overwrite existing rule?</th>
+      <th>Output</th>
+    </tr>
+  </thead>
+  <tbody>
+
+  
+  <tr>
+    <td>Not installed</td>
+    <td>No</td>
+    <td>N/A</td>
+    <td>
+<pre>
+{
+  name: "My custom rule",
+} 
+</pre>
+    </td>
+    <td>N/A</td>
+    <td>
+<pre>
+{
+  name: "My custom rule",
+  rule_source: {
+    type: "internal"
+  }
+} 
+</pre>
+      </td>
+    </tr>
+  
+
+
+  <tr>
+    <td>
+<pre>
+{
+  name: "My custom rule",
+} 
+</pre>
+    </td>
+    <td>No</td>
+    <td>N/A</td>
+    <td>
+<pre>
+{
+  name: "My edited rule",
+} 
+</pre>
+    </td>
+    <td>
+      <b>No</b>
+    </td>
+    <td>
+      Rule import rejected<br> because of rule_id match
+    </td>
+    <tr>
+
+
+  <tr>
+    <td>
+<pre>
+{
+  name: "My custom rule",
+} 
+</pre>
+    </td>
+    <td>No</td>
+    <td>N/A</td>
+    <td>
+<pre>
+{
+  name: "My edited rule",
+} 
+</pre>
+    </td>
+    <td>
+      Yes
+    </td>
+    <td>
+<pre>
+{
+  name: "My edited rule",
+  rule_source: {
+    type: "internal"
+  }
+} 
+</pre>
+      </td>
+    <tr>
+
+
+  <tr>
+    <td>
+      Not installed
+    </td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+  source_updated_at: "2024-05-..."
+} 
+</pre>
+    </td>
+    <td><b>No</b></td>
+    <td>
+<pre>
+{
+  name: "My prebuilt rule",
+} 
+</pre>
+    </td>
+    <td>N/A</td>
+    <td>
+      Prebuilt rule import rejected because no matching <br>
```

Comment:
I am not sure about this scenario - it seems to be something discussed in https://github.com/elastic/kibana/pull/171856#discussion_r1524431230, but it seems not to be updated in the table here.

Reply:
Thanks, I missed updating this row. Updated now